task/orchestrator_test.go

```diff
@@ -84,6 +87,30 @@ func TestOrchestrator(t *testing.T) {
 			gracePeriod: 2 * time.Second,
 			wantError:   "",
 		},
+		{
+			name:   "child processes are cleaned up",
```

Comment:
Added a test to ensure the child process group is indeed terminated on shutdown of the agent.